src/distinguished_name.cpp

```diff
@@ -84,22 +92,32 @@ DistinguishedName DistinguishedName::fromX509Name(X509_NAME *x509)
     return builder.build();
 }
 
-auto _createTuple(const DistinguishedName &dn)
+auto _createTuple(const DistinguishedName& dn)
 {
-    return std::tie(dn.commonName(),
+    return std::make_tuple(dn.commonName(),
```

Comment:
This is used in the comparison operator, but the comparison operator now does not take the order into account. I would expect the following behavior:

 - Comparing a non-order-aware with a non-order-aware DN: previous behavior
 - Comparing an order-aware with a non-order-aware DN (and vice-versa): Only equal if the order-aware entries are in the default order and the contents match.
 - Comparing two order-aware DNs: Only equal if the order and the contents match.

Reply:
Sure we can check the order using the CustomOrderBuilder. However if the DN is created from a X509 then the order is not preserved and will then also not matter for matching. Martin and I talked about it and it is quite an effort to add the missing API from openssl to iterate over the attributes and therefore record the actual order. This is can be done in a second step which is out of the scope from the pull request.